### ISSUE TITLE
Contrastive Loss Function

### ISSUE THREAD

**Issue by old-school-kid** [keras-team/keras#18203]
**System information**.

TensorFlow version (you are using): 2.12
Are you willing to contribute it (Yes/No) :  Yes

**Describe the feature and the current behavior/state**.
Pair loss is used in contrastive pre-training (backbone of CLIP) and siamese network which is kind of very important in the field of computer vision and multi modal learning. We dont have to create pairwise data for contrastive loss, rather can take samples from minibatches. Also if we want to add this feature should we directly add the contrastive loss (hardcode to only single label) or use more of a n-pair loss where the logit is calculated by the user and plugged into the loss function?

**Papers**
[CLIP](https://arxiv.org/abs/2103.00020) Citations: 5558
[Siamese Network](https://www.cs.cmu.edu/~rsalakhu/papers/oneshot1.pdf) Citations: 4135

**Will this change the current api? How?**
Yes. We will just add one more loss function.

**Who will benefit from this feature?**
Everyone who uses Keras and uses it for model pre-training or even contrastive learning.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):
[Tensorflow addon implementation](https://github.com/tensorflow/addons/blob/v0.20.0/tensorflow_addons/losses/contrastive.py)

**Comment by innat** [comment]
https://github.com/search?q=repo%3Atensorflow%2Fsimilarity+Contrastive+Loss&type=code
